### PR TITLE
Working Splunk Server and Fowarder

### PR DESCRIPTION
**Pull request by tspeigner**

### CODE REVIEW

**Comment by tspeigner** [comment]
@maju6406 will you check this one out now?  got rid of the firewall changes to the fileserver.

File: site/profile/manifests/app/tse_master.pp [hunk 1]
```diff
@@ -62,4 +62,5 @@
     source => 'puppet:///modules/profile/puppet-code-commit',
   }
```
Comment (maju6406) [inline site/profile/manifests/app/tse_master.pp]: We should probably take this out of the PR too.

File: site/profile/manifests/app/tse_master/files/splunk.pp [hunk 2]
```diff
@@ -1,88 +1,88 @@
-class profile::app::tse_master::files::splunk (
```
Comment (maju6406) [inline site/profile/manifests/app/tse_master/files/splunk.pp]: This probably shouldn't be in the PR either right? Would it better to just delete this file?

File: site/profile/manifests/platform/monitoring/splunkforwarder.pp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  
+  # This is section forwards the system Audit log
+  @splunkforwarder_input { "${hostname}_audit" :
+    section => 'monitor:///var/log/audit/audit.log',
+    setting => 'sourcetype',
+    value   => "${hostname}_audit",
+    tag     => 'splunk_forwarder'
+  }
+include ::splunk::forwarder
```
Comment (maju6406) [inline site/profile/manifests/platform/monitoring/splunkforwarder.pp]: Dont think you need this include anymore. It was there for testing.

File: site/profile/manifests/app/splunk/server.pp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  # splunkforwarder manifest for setting the host value.
+  # A "poor persons" DNS fix.
+
+    @@host { $facts['fqdn'] :
+      comment      => 'Splunk Server',
+      ip           => $facts['ipaddress'],
+      tag          => 'splunkserver',
+    } 
+
+include splunk
```
Comment (maju6406) [inline site/profile/manifests/app/splunk/server.pp]: Dont think you need this include anymore. It was there for testing.

File: site/profile/manifests/platform/monitoring/splunkforwarder.pp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    $_splunk_server = $splunk_server
+  }
+
+  class { '::splunk::params':
+    server   => $_splunk_server,
+    src_root => 'splunk::params::src_root',
+  }
+  # Splunkforwarder input setup.  This is just collecting the /var/log/messages
+  # from each node and shipping its logs to the splunk server
+  @splunkforwarder_input { "${hostname}_messages" :
```
Comment (maju6406) [inline site/profile/manifests/platform/monitoring/splunkforwarder.pp]: Do these splunkforwarder_input need to be virtual resources? Why?

File: site/profile/manifests/platform/monitoring/splunkforwarder.pp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    section => 'monitor:///var/log/messages',
+    setting => 'sourcetype',
+    value   => "${hostname}_messages",
+    tag     => 'splunk_forwarder'
+  }
+  
+  # This is section forwards the system Audit log
+  @splunkforwarder_input { "${hostname}_audit" :
```
Comment (maju6406) [inline site/profile/manifests/platform/monitoring/splunkforwarder.pp]: Do these splunkforwarder_input need to be virtual resources? Why?

File: site/profile/manifests/app/splunk/server.pp [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# splunk server module
+class profile::app::splunk::server (
+  $purge_inputs       = false,
```
Comment (maju6406) [inline site/profile/manifests/app/splunk/server.pp]: Don't think you need either purge_inputs and purge_outputs parameters. I'd say that even the src_root is unecessary.

File: site/profile/manifests/app/splunk/README.md [hunk 8]
```diff
@@ -0,0 +1,2 @@
+# Sets up Splunk Server - started 6/15/2017 - TLS
+# Specify settings in the module to start with, with plans to move it to hiera
```
Comment (maju6406) [inline site/profile/manifests/app/splunk/README.md]: Don't know if this README makes sense to include. Maybe rewrite it to be more useful?

File: site/profile/manifests/platform/monitoring/splunkforwarder.pp [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    $splunk_nodes_query = 'resources[certname] { type = "Class" and title = "Splunk" }'
+    $_splunk_server = puppetdb_query($splunk_nodes_query)[0][certname]
+    Host  <<| tag == 'splunkserver' |>>
+  } else {
+    $_splunk_server = $splunk_server
+  }
+
+  class { '::splunk::params':
+    server   => $_splunk_server,
+    src_root => 'splunk::params::src_root',
```
Comment (maju6406) [inline site/profile/manifests/platform/monitoring/splunkforwarder.pp]: Don't think this src_root value works. You're setting it to the literal string 'splunk::params::src_root'. Is it supposed to be $splunk::params:src_root or ::splunk::params:src_root?

File: site/profile/manifests/app/tse_master/files/splunk.pp (hunk 2 above)
Comment (tspeigner) [inline site/profile/manifests/app/tse_master/files/splunk.pp]: I'll just get rid of that file and redo the PR.

File: site/profile/manifests/platform/monitoring/splunkforwarder.pp (hunk 5 above)
Comment (tspeigner) [inline site/profile/manifests/platform/monitoring/splunkforwarder.pp]: In order to start sending some log data, users can take advantage of the Splunkforwarder_input type. Here is a basic example of adding an input to start sending Puppet Server logs:

https://forge.puppet.com/puppet/splunk#usage

This virtual resource will get collected by the ::splunk::forwarder class if it is tagged with splunk_forwarder and will add the appropriate setting to the inputs.conf file and refresh the service.

File: site/profile/manifests/platform/monitoring/splunkforwarder.pp (hunk 6 above)
Comment (tspeigner) [inline site/profile/manifests/platform/monitoring/splunkforwarder.pp]: This virtual resource will get collected by the ::splunk::forwarder class if it is tagged with splunk_forwarder and will add the appropriate setting to the inputs.conf file and refresh the service.

**Pull request closed without merging**